### ISSUE TITLE
add compatibility for custom metrics

### ISSUE THREAD

**Issue by gsicard** [gsicard/torch-async#7]
As of now, only training loss and validation loss and accuracy are computed during training. Future version should support custom metrics and possibly PyTorch Lightning's torchmetrics.
